modules/sfp_phishstats.py

```diff
@@ -0,0 +1,186 @@
+          headers=headers,
+          timeout=15,
+          useragent=self.opts['_useragent']
+        )
+
+        if not res['code'] == "200":
+            self.sf.debug("No information found from Phishstats for IP Address")
+            return None
+
+        try:
+            return json.loads(res['content'])
+        except:
+            return None
+         
+    # Handle events sent to this module
+    def handleEvent(self, event):
+        
+        eventName = event.eventType
+        srcModuleName = event.module
+        eventData = event.data
+        
+        if self.errorState:
+            return None
+
+        self.sf.debug("Received event, " + eventName + ", from " + srcModuleName)
+
+        # Don't look up stuff twice
+        if eventData in self.results:
+            self.sf.debug("Skipping " + eventData + " as already mapped.")
+            return None
+        else:
+            self.results[eventData] = True
+                
+        if eventName == 'NETBLOCK_OWNER':
+            if not self.opts['netblocklookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxnetblock']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxnetblock']))
+                    return None
+        
+        if eventName == 'NETBLOCK_MEMBER':
+            if not self.opts['subnetlookup']:
+                return None
+            else:
+                if IPNetwork(eventData).prefixlen < self.opts['maxsubnet']:
+                    self.sf.debug("Network size bigger than permitted: " +
+                                str(IPNetwork(eventData).prefixlen) + " > " +
+                                str(self.opts['maxsubnet']))
+                    return None
+                    
+        qrylist = list()
+        if eventName.startswith("NETBLOCK_"):
+            for ipaddr in IPNetwork(eventData):
+                qrylist.append(str(ipaddr))
+                self.results[str(ipaddr)] = True
+        else:
+            # If user has enabled affiliate checking
+            if eventName == "AFFILIATE_IPADDR" and not self.opts['checkaffiliates']:
+                return None
+            qrylist.append(eventData)
+        
+        for addr in qrylist:
+
+            if self.checkForStop():
+                return None
+            
+            data = self.queryIPAddress(addr)
+
+            if data is None:
+                break
+            
+            try:
+                maliciousIP = data[0].get('ip')
+            except:
+                # If ArrayIndex is out of bounds then data doesn't exist
+                continue
+        
+            if maliciousIP is None:
+                continue
+
+            if addr != maliciousIP:
+                self.sf.error("Reported address doesn't match requested, skipping")
```

Comment:
If you don't set `False` as the second argument, this becomes an error that aborts the scan.